distribution/resources/bin/gateway

```diff
@@ -45,30 +45,106 @@ PRGDIR=`dirname "$PRG"`
 
 # set BALLERINA_HOME
 GWHOME=`cd "$PRGDIR/.." ; pwd`
-export BALLERINA_HOME=$GWHOME'/runtime'
 export JAVA_HOME=$GWHOME'/lib/jdk8u202-b08-jre'
 export MGW_VERSION="3.1.0"
 export GW_HOME="$GWHOME"
 
-if [ -e "$GWHOME/bin/gateway.pid" ]; then
-  PID=`cat "$GWHOME"/bin/gateway.pid`
+JAVA_PATH=$GWHOME/lib/jdk8u202-b08-jre
+if test -d "$JAVA_PATH"; then
+  JAVA_HOME=$JAVA_PATH
 fi
 
-#To run the Java classes
-java_cmd () {
-  if [ -z "$JAVACMD" ] ; then
-      if [ -n "$JAVA_HOME"  ] ; then
-          if [ -x "$JAVA_HOME/jre/sh/java" ] ; then
-              # IBM's JDK on AIX uses strange locations for the executables
-              JAVACMD="$JAVA_HOME/jre/sh/java"
-          else
-              JAVACMD="$JAVA_HOME/bin/java"
-          fi
-      else
+BASE_DIR=$PWD
+cygwin=false;
+darwin=false;
+os400=false;
+mingw=false;
+case "`uname`" in
+CYGWIN*) cygwin=true;;
+MINGW*) mingw=true;;
+OS400*) os400=true;;
+Darwin*) darwin=true
+esac
+
+# For Cygwin, ensure paths are in UNIX format before anything is touched
+if $cygwin; then
+  [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
+fi
+
+# For OS400
+if $os400; then
+  # Set job priority to standard for interactive (interactive - 6) by using
+  # the interactive priority - 6, the helper threads that respond to requests
+  # will be running at the same priority as interactive jobs.
+  COMMAND='chgjob job('$JOBNAME') runpty(6)'
+  system $COMMAND
+
+  # Enable multi threading
+  QIBM_MULTI_THREADED=Y
+  export QIBM_MULTI_THREADED
+fi
+
+# For Migwn, ensure paths are in UNIX format before anything is touched
+if $mingw ; then
+  [ -n "$JAVA_HOME" ] &&
+    JAVA_HOME="`(cd "$JAVA_HOME"; pwd)`"
+fi
+
+if [ -z "$JAVACMD" ] ; then
+    if [ -n "$JAVA_HOME"  ] ; then
+        if [ -x "$JAVA_HOME/jre/sh/java" ] ; then
+            # IBM's JDK on AIX uses strange locations for the executables
+            JAVACMD="$JAVA_HOME/jre/sh/java"
+        else
+            JAVACMD="$JAVA_HOME/bin/java"
+        fi
+    else
           JAVACMD=java
+    fi
+fi
+
+# For Cygwin, switch paths to Windows format before running java
+if $cygwin; then
+    JAVACMD=`cygpath --absolute --windows "$JAVACMD"`
+fi
+
+if [ ! -x "$JAVACMD" ] ; then
+  echo "Error: JAVA Package inside the microgateway runtime distribution is not working."
+  exit 1
+fi
+
+# Ballerina test/debug command
+argsLength=$#
+args=($@)
+for (( i=0; i<argsLength; i++ )); do
+  if [  "${args[$i]}" = "test" ]; then
+    for (( j=0; j<argsLength; j++ )); do
+      if [ "${args[$j]}" = "--debug" ]; then
+        BAL_JAVA_DEBUG=${args[$j+1]}
       fi
+    done
   fi
+done
+
+if [ -n "${BAL_JAVA_DEBUG+set}" ]; then
+    if [ -z "$BAL_JAVA_DEBUG" ]; then
+        echo "Please specify the debug port for the BAL_JAVA_DEBUG variable"
+  exit 1
+    else
```

Comment:
Check the indentation of else block

Reply:
added with https://github.com/wso2/product-microgateway/pull/944/commits/ee002a57923994065b743410db14e25ac065a88b